api/src/test/java/io/opentelemetry/trace/attributes/SemanticAttributesTest.java

```diff
@@ -65,6 +67,35 @@ public void shouldEnableSetAttributeOnSpan() throws IllegalAccessException {
     assertThat(keys.size()).isEqualTo(fields.length);
   }
 
+  @Test
+  public void shouldEnableSetAttributeOnSpanBuilder() throws IllegalAccessException {
+    Set<String> keys = new HashSet<>();
+    Field[] fields = SemanticAttributes.class.getFields();
+    for (Field field : fields) {
+      Object attribute = field.get(null);
+      if (attribute instanceof StringAttributeSetter) {
+        keys.add(((StringAttributeSetter) attribute).key());
+        spanBuilder.apply(((StringAttributeSetter) attribute).set("TestValue"));
+        spanBuilder.apply(((StringAttributeSetter) attribute).set(null));
+      } else if (attribute instanceof LongAttributeSetter) {
+        keys.add(((LongAttributeSetter) attribute).key());
+        spanBuilder.apply(((LongAttributeSetter) attribute).set(42));
+      } else if (attribute instanceof DoubleAttributeSetter) {
+        keys.add(((DoubleAttributeSetter) attribute).key());
+        spanBuilder.apply(((DoubleAttributeSetter) attribute).set(3.14));
+      } else if (attribute instanceof BooleanAttributeSetter) {
+        keys.add(((BooleanAttributeSetter) attribute).key());
+        spanBuilder.apply(((BooleanAttributeSetter) attribute).set(true));
+      }
```

Comment:
maybe add an `else` that calls `fail`, so that this test will highlight something not being covered?

Reply:
IMHO the test for both this and the old SemConv API needs quite some improvements (such as testing that `setAttribute` was actually called, which is unfortunately not possible with only the API due to Mockito not supporting final classes). But I just copied & adapted the tests for the `set` method for now.

Reply:
I'd be all in favor of making those classes non-final!